packages/babel-plugin-redux-saga/src/index.js

```diff
@@ -1,75 +1,40 @@
 var SourceMapConsumer = require('source-map').SourceMapConsumer
 var pathFS = require('path')
-var importHelper = require('@babel/helper-module-imports')
 
-var fnId = 'reduxSagaSource'
-var tempVarId = 'res'
-var locationSymbolPath = 'redux-saga'
-var locationSymbolExportName = 'SAGA_LOCATION'
-var hintedLocationSymbolExportName = locationSymbolExportName
+var globalSymbolName = '@@redux-saga/LOCATION'
 
 module.exports = function(babel) {
-  var { types: t } = babel
+  var { types: t, template } = babel
   var sourceMap = null
-  var symbolNameBinding = null
 
-  function getAssignementWithSymbolProperty(name, symbolName) {
-    return t.memberExpression(t.identifier(name), t.identifier(symbolName), true)
-  }
-
-  function getObjectExtensionNode(path, functionName) {
-    if (!symbolNameBinding) {
-      const { name: locationSymbolName } = importHelper.addNamed(path, locationSymbolExportName, locationSymbolPath, {
-        nameHint: hintedLocationSymbolExportName,
-      })
-
-      symbolNameBinding = locationSymbolName
-    }
-    return getAssignementWithSymbolProperty(functionName, symbolNameBinding)
-  }
-
-  //  eslint-disable-next-line no-unused-vars
-  function getParentData(path, state) {
-    var parent = path.findParent(path => path.isFunction())
-    var locationData = calcLocation(parent.node.loc, state.file.opts.filename, state.opts.basePath)
-    var name = parent.node.id ? parent.node.id.name : 'λ'
-    return {
-      name,
-      location: locationData,
-    }
-  }
-
-  function assignLoc(objectExtensionNode, fileName, lineNumber) {
-    return t.expressionStatement(
-      t.assignmentExpression(
-        '=',
-        objectExtensionNode,
-        t.objectExpression([
-          t.objectProperty(t.identifier('fileName'), t.stringLiteral(fileName)),
-          t.objectProperty(t.identifier('lineNumber'), t.numericLiteral(lineNumber)),
-        ]),
-      ),
-    )
-  }
-
-  function wrapIIFE(node, objectExtension, fileName, lineNumber, sourceCode) {
-    const body = [
-      t.variableDeclaration('var', [t.variableDeclarator(t.identifier(tempVarId), node)]),
-      t.expressionStatement(
-        t.assignmentExpression(
-          '=',
-          objectExtension,
-          t.objectExpression([
-            t.objectProperty(t.identifier('fileName'), t.stringLiteral(fileName)),
-            t.objectProperty(t.identifier('lineNumber'), t.numericLiteral(lineNumber)),
-            t.objectProperty(t.identifier('code'), t.stringLiteral(sourceCode)),
-          ]),
-        ),
-      ),
-      t.returnStatement(t.identifier(tempVarId)),
-    ]
-    const container = t.functionExpression(t.identifier(fnId), [], t.blockStatement(body))
-    return t.callExpression(container, [])
+  const extendExpressionWithLocation = template(`
+    (function reduxSagaSource() {
+      return Object.defineProperty(TARGET, SYMBOL_NAME, {
```

Comment:
When using real symbols we don't have to use `Object.defineProperty`, but using it consistently makes the code easier, so let's leave it that way for now.

Reply:
yes, that was done intentionally to unify approach